pkg/syncer/metadatasyncer.go

```diff
@@ -1041,6 +1043,15 @@ func csiPVDeleted(ctx context.Context, pv *v1.PersistentVolume, metadataSyncer *
 
 		if err := metadataSyncer.volumeManager.DeleteVolume(ctx, volumeHandle, deleteDisk); err != nil {
 			log.Errorf("PVDeleted: Failed to delete disk %s with error %+v", volumeHandle, err)
+			return
+		}
+		if migrationFeatureEnabled && pv.Spec.VsphereVolume != nil {
```

Comment:
We do not need to check ` pv.Spec.VsphereVolume != nil` here.

Reply:
Shouldn't we need this to avoid invoking `volumeMigrationService.DeleteVolumeInfo` for CSI volumes and when migration feature state is true

Reply:
yes. keep it then.